pkg/config/validation/validation.go

```diff
@@ -1936,6 +1939,30 @@ func ValidateControlPlaneAuthPolicy(policy meshconfig.AuthenticationPolicy) erro
 	return fmt.Errorf("unrecognized control plane auth policy %q", policy)
 }
 
+func validatePolicyTargetReference(targetRef *type_beta.PolicyTargetReference) (v Validation) {
+	if targetRef == nil {
+		return
+	}
+	if targetRef.Name == "" {
+		v = appendErrorf(v, "policyTargetReference.name must be set")
+	}
+	// TODO: we should also validate that the namespace is not a root namespace
+	if targetRef.Namespace == nil {
+		v = appendErrorf(v, "policyTargetReference.namespace must be set")
+	}
+	// Currently, gateway.networking.k8s.io is the only valid Group for this field.
+	if targetRef.Group != defaultGatewayGroup {
+		v = appendErrorf(v, "policyTargetReference.group is incorrect; expected: %v, got: %v", defaultGatewayGroup, targetRef.Group)
+	}
+	// Currently, gateway.networking.k8s.io/Gateway is the only valid Kind for this field.
+	if targetRef.Kind != defaultGatewayKind {
+		v = appendErrorf(v, "policyTargetReference.kind is incorrect; expected: %v, got: %v", defaultGatewayKind, targetRef.Kind)
+	}
+	return
+}
+
+// if you have a layer7 policy with no targetRef then it will always be invalid
```

Comment:
I think it is if you have a non layer 7 policy with no targetRef then it is invalid. But for now we are just falling back to workload selectors.

Reply:
Ahh yeah you're right, this is a premature comment, I'll remove it for now. But once [layer targeted authorization policy RFC](https://docs.google.com/document/d/1EDnk9cyji8GtpK7tF6czqiStd7UTujoIhhBUi1XYAQM/edit#heading=h.stll71jzvhaf) is approved then `if you have a layer7 policy with no targetRef then it will always be invalid` will be true.